File: book_store.rs
// i hate this one
use std::cmp::min;
const PRICES : [u32;6] = [0,800,1520, 2160, 2560, 3000];
pub fn lowest_price(books: &[u32]) -> u32 {
    let mut v = vec![(0,1),(0,2),(0,3),(0,4),(0,5)];
    for b in books{
        v[(b - 1) as usize].0 += 1;
    }
    custom_sort(&mut v);
    f(v)
}
fn f(v : Vec<(u32,u8)>) -> u32{
    let mut cost = v.iter().map(|(c,_)| c).sum::<u32>() * 800;
    for i in 2..=v.iter().filter(|&&(c,_)| c > 0).count(){
        let mut v_c = v.clone();
        for j in 0..i{
            v_c[j].0 -= 1;
        }
        custom_sort(&mut v_c);
        cost = min(cost , PRICES[i] + f(v_c));
    }
    cost
}
fn custom_sort(arr : &mut Vec<(u32,u8)>){
    // reverse sorting by occurances
    arr.sort_unstable_by(|(a,_),(b,_)| b.cmp(a));
}

File: solutions/python/book-store/1/book_store.py
from math import inf
PRICES = [0, 800, 1520, 2160, 2560, 3000]
def total(books):
    v = [(0, i) for i in range(1, 6)]
    for b in books:
        v[b - 1] = (v[b - 1][0] + 1, v[b - 1][1])
    custom_sort(v)
    return f(v)

def f(v):
    cost = sum(c for c, _ in v) * 800
    distinct = sum(1 for c, _ in v if c > 0)
    for i in range(2, distinct + 1):
        v_c = v.copy()
        for j in range(i):
            c, book_id = v_c[j]
            v_c[j] = (c - 1, book_id)
        custom_sort(v_c)
        cost = min(cost, PRICES[i] + f(v_c))

    return cost

def custom_sort(arr):
    arr.sort(key=lambda x: -x[0])
